pootle/apps/pootle_project/models.py

```diff
@@ -510,6 +524,24 @@ def _get_code(self, project):
     ### /TreeItem
 
 
+@receiver([vfolder_post_save, pre_delete])
+def invalidate_resources_cache_for_vfolders(sender, instance, **kwargs):
+    if instance.__class__.__name__ == 'VirtualFolder':
+        try:
+            # In case this is vfolder_post_save.
+            affected_projects = kwargs['projects']
+        except KeyError:
+            # In case this is pre_delete.
+            affected_projects = Project.objects.filter(
+                translationproject__stores__unit__vfolders=instance
+            ).distinct().values_list('code', flat=True)
+
+        cache.delete_many([
+            make_method_key('Project', 'resources', proj)
+            for proj in affected_projects
```

Comment:
Honestly I don't think it's worth the complexity (and duplication) of calculating the affected projects: the invalidation is triggered for every vfolder save/delete, it might perfectly be going through several projects already, and calculating the resources is not expensive. Besides, the `vfolder_post_save` would be irrelevant here (there's a duck typing check in place already).

So I'd appreciate if you can use the approach I already proposed in this PR.